### ISSUE TITLE
Extend the tests in regexp/inspect_spec with over escaping combined with character classes

### ISSUE THREAD

**Issue by herwinw** [ruby/spec#987]
The specs have one test for escaped backslashes in `Regexp#inspect`:
```ruby
it "does not over escape" do
  Regexp.new('\\\/').inspect.should == "/\\\\\\//"
end
```
The obvious implementation to make this spec pass is to simply escape every slash with another slash. This works in this case, but that causes an error in `language/case_spec` for something that is not related to the case statement:
```ruby
it "tests with a regexp interpolated within another regexp" do
  digits_regexp = /\d+/
  case "foo43"
  when /oo(#{digits_regexp})/
```
This results in a regexp `/oo(?-mix:\\d+)/`, which now expects a slash followed by one or more d characters. This does not match the input, but does not indicate the escaping as the issue.

I tested my implementation with the these specs added, but this was more or less brute forcing every possibility. Also, because `Regexp.new` takes a String which has an additional layer of escaping, I peferred the `//` notation.
```
Regexp.new('\d+').inspect.should == "/\\d+/"
/\d+/.inspect.should == "/\\d+/"

Regexp.new('\\d+').inspect.should == "/\\d+/"
/\\d+/.inspect.should == "/\\\\d+/"

Regexp.new('\\\d+').inspect.should == "/\\\\d+/"
/\\\d+/.inspect.should == "/\\\\\\d+/"

Regexp.new('\\\\d+').inspect.should == "/\\\\d+/"
/\\\\d+/.inspect.should == "/\\\\\\\\d+/"
```

I think it would be better to at least include the following cases:
```ruby
/\d+/ # Output should have 1 not escaped slash, since this should not be escaped
/\\d+/ # Output should have 1 escaped slash (so 2 slashes), this should be escaped
/\\\d+/ # Output should have 1 escaped and 1 not escaped slash (so 3 slashes)
```

Furthermore, I think it would be better to add an explicit test for the regexp embedded in regexp scenario like there is in the case spec now, just to make the test a bit more explicit and decouple it from the case statement.

I'm not that familiar with this project and how things are structured, can I just add a few tests to the inspect spec of regexp, and make a new spec file for the embedding tests?

**Comment by herwinw** [comment]
For what it's worth: https://github.com/natalie-lang/natalie/pull/779 has a bit of a write up with the issues I encountered

**Comment by eregon** [comment]
> I'm not that familiar with this project and how things are structured, can I just add a few tests to the inspect spec of regexp, and make a new spec file for the embedding tests?

Yes, just make a PR and based on the changes we can find out what's the right place.
Specs testing interpolating a regexp in another regexp should probably move to `language/regexp_spec.rb`.
